### PR TITLE
fix(trace-view): Do not recreate fuse in trace view

### PR DESCRIPTION
**Pull request by Zylphrex**
Stop recreating a new instance of fuse each time a search is started.

### CODE REVIEW

File: static/app/views/performance/traceDetails/content.tsx [hunk 1]
```diff
@@ -91,6 +92,46 @@ class TraceDetailsContent extends React.Component<Props, State> {
     filteredTransactionIds: undefined,
   };
 
+  componentDidUpdate(prevProps: Props) {
```
Comment (dashed) [inline static/app/views/performance/traceDetails/content.tsx:95]: `componentDidUpdate()` will not be called for the initial render. Should there be something called for `componentDidMount()`?

File: static/app/views/performance/traceDetails/content.tsx [hunk 2]
```diff
@@ -190,37 +236,7 @@ class TraceDetailsContent extends React.Component<Props, State> {
...
-      keys: ['indexed'],
-      includeMatches: true,
-      threshold: 0.6,
-      location: 0,
-      distance: 100,
-      maxPatternLength: 32,
-    });
-
-    const fuseMatches = fuse
+    const fuseMatches = this.fuse
```
Comment (dashed) [inline static/app/views/performance/traceDetails/content.tsx:239]: @Zylphrex On line 226 (unable to comment there), `filterTransactions()` method shouldn't need to have an `async` keyword now that the asynchronous code has been moved to `async initFuse()`.

Comment (Zylphrex) [reply static/app/views/performance/traceDetails/content.tsx:239]: removed in 15cbcaf

**Review by dashed — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**